activation/activation.go

```
@@ -203,31 +203,47 @@ func (b *Builder) PublishActivationTx(epoch types.EpochId) error {
 
 		hash, err := b.challenge.Hash()
 		if err != nil {
-			return fmt.Errorf("getting challenge hash failed: %v", err)
+			return false, fmt.Errorf("getting challenge hash failed: %v", err)
 		}
 		b.nipst, err = b.nipstBuilder.BuildNIPST(hash)
 		if err != nil {
-			return fmt.Errorf("cannot create nipst " + err.Error())
+			return false, fmt.Errorf("cannot create nipst: %v", err)
 		}
 	}
 	if b.mesh.LatestLayer().GetEpoch(b.layersPerEpoch) < b.challenge.PubLayerIdx.GetEpoch(b.layersPerEpoch) {
-		return fmt.Errorf("an epoch has not passed during nipst creation, current: %v wanted: %v", b.mesh.LatestLayer().GetEpoch(b.layersPerEpoch), b.challenge.PubLayerIdx.GetEpoch(b.layersPerEpoch))
+		b.log.Warning("an epoch has not passed during nipst creation, current: %v, mesh: %v, wanted: %v, waiting",
+			epoch, b.mesh.LatestLayer().GetEpoch(b.layersPerEpoch), b.challenge.PubLayerIdx.GetEpoch(b.layersPerEpoch))
+		return false, nil
 	}
 
 	// when we reach here an epoch has passed
 	// we've completed the sequential work, now before publishing the atx,
 	// we need to provide number of atx seen in the epoch of the positioning atx.
-	activeIds := uint32(b.activeSet.ActiveSetSize(b.posLayerID.GetEpoch(b.layersPerEpoch)))
-	b.log.Info("active ids seen for epoch of the pos atx (epoch: %v) is %v", b.posLayerID.GetEpoch(b.layersPerEpoch), activeIds)
-	atx := types.NewActivationTxWithChallenge(*b.challenge, activeIds, b.mesh.GetLatestView(), b.nipst, true)
+	activeIds := b.activeSet.ActiveSetSize(b.posLayerID.GetEpoch(b.layersPerEpoch))
```

consider changing ActiveSetSize to return an error instead of returning 0